File: my project/my project/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private TextView txtShow;
    private Button btnZero, btnOne, btnTwo, btnThree, btnFour, btnFive, btnSix, btnSeven, btnEight, btnNine, btnStar, btnClear, btnCall;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // 連結 XML 元件
        txtShow = findViewById(R.id.textView);
        btnZero = findViewById(R.id.btnZero);
        btnOne = findViewById(R.id.btnOne);
        btnTwo = findViewById(R.id.btnTwo);
        btnThree = findViewById(R.id.btnThree);
        btnFour = findViewById(R.id.btnFour);
        btnFive = findViewById(R.id.btnFive);
        btnSix = findViewById(R.id.btnSix);
        btnSeven = findViewById(R.id.btnSeven);
        btnEight = findViewById(R.id.btnEight);
        btnNine = findViewById(R.id.btnNine);
        btnStar = findViewById(R.id.btnStar);
        btnClear = findViewById(R.id.btnClear);
        btnCall = findViewById(R.id.btnCall);   // 新增 CALL 按鈕

        // 所有按鈕共用監聽器
        View.OnClickListener myListener = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String s = txtShow.getText().toString();
                int id = v.getId();

                if (id == R.id.btnZero) {
                    txtShow.setText(s + "0");
                } else if (id == R.id.btnOne) {
                    txtShow.setText(s + "1");
                } else if (id == R.id.btnTwo) {
                    txtShow.setText(s + "2");
                } else if (id == R.id.btnThree) {
                    txtShow.setText(s + "3");
                } else if (id == R.id.btnFour) {
                    txtShow.setText(s + "4");
                } else if (id == R.id.btnFive) {
                    txtShow.setText(s + "5");
                } else if (id == R.id.btnSix) {
                    txtShow.setText(s + "6");
                } else if (id == R.id.btnSeven) {
                    txtShow.setText(s + "7");
                } else if (id == R.id.btnEight) {
                    txtShow.setText(s + "8");
                } else if (id == R.id.btnNine) {
                    txtShow.setText(s + "9");
                } else if (id == R.id.btnStar) {
                    txtShow.setText(s + "*");
                } else if (id == R.id.btnClear) {
                    txtShow.setText("電話號碼:");
                } else if (id == R.id.btnCall) {
                    // 取得輸入的號碼
                    String number = txtShow.getText().toString();
                    if (!number.isEmpty()) {
                        // 建立撥號 Intent
                        Intent intent = new Intent(Intent.ACTION_DIAL);
                        intent.setData(Uri.parse("tel:" + number));
                        startActivity(intent);
                    }
                }
            }
        };


        btnZero.setOnClickListener(myListener);
        btnOne.setOnClickListener(myListener);
        btnTwo.setOnClickListener(myListener);
        btnThree.setOnClickListener(myListener);
        btnFour.setOnClickListener(myListener);
        btnFive.setOnClickListener(myListener);
        btnSix.setOnClickListener(myListener);
        btnSeven.setOnClickListener(myListener);
        btnEight.setOnClickListener(myListener);
        btnNine.setOnClickListener(myListener);
        btnStar.setOnClickListener(myListener);
        btnClear.setOnClickListener(myListener);
        btnCall.setOnClickListener(myListener);
    }
}

File: Homework3.4_fast food/app/src/main/java/com/example/meal/MainActivity.kt
package com.example.meal

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    companion object {
        var mainMeal: String? = null
        var sideDish: String? = null
        var drink: String? = null
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // 取得元件
        val txtOrder = findViewById<TextView>(R.id.txtOrder)
        val btnMainMeal = findViewById<Button>(R.id.btnMainMeal)
        val btnSideDish = findViewById<Button>(R.id.btnSideDish)
        val btnDrink = findViewById<Button>(R.id.btnDrink)
        val btnConfirm = findViewById<Button>(R.id.btnConfirm)

        // 點擊進入各畫面
        btnMainMeal.setOnClickListener {
            startActivity(Intent(this, MainMealActivity::class.java))
        }

        btnSideDish.setOnClickListener {
            startActivity(Intent(this, SideDishActivity::class.java))
        }

        btnDrink.setOnClickListener {
            startActivity(Intent(this, DrinkActivity::class.java))
        }

        btnConfirm.setOnClickListener {
            startActivity(Intent(this, ConfirmActivity::class.java))
        }

        // 更新目前選擇（onCreate 只顯示初始值）
        val summary = StringBuilder()
        if (mainMeal != null) summary.append("主餐: $mainMeal\n")
        if (sideDish != null) summary.append("副餐: $sideDish\n")
        if (drink != null) summary.append("飲料: $drink\n")
        txtOrder.text = if (summary.isEmpty()) "尚未點餐" else summary.toString()
    }

    override fun onResume() {
        super.onResume()

        // 回來主頁時更新文字
        val txtOrder = findViewById<TextView>(R.id.txtOrder)
        val summary = StringBuilder()
        if (mainMeal != null) summary.append("主餐: $mainMeal\n")
        if (sideDish != null) summary.append("副餐: $sideDish\n")
        if (drink != null) summary.append("飲料: $drink\n")

        txtOrder.text = if (summary.isEmpty()) "尚未點餐" else summary.toString()
    }
}

